### PR TITLE
Dev hub redesign

### PR DESCRIPTION
**Pull request by zesuma**
## Description & motivation
There are still a few blockers we need to work out before this can go live but I wanted to get a preview link for Jason to see what I need his advice on.

ASANA ticket
https://app.asana.com/0/1200099998847559/1202844016314421/f

## To-do before merge
**Blockers**
1. **LOGO on DOCS Page:** We Still need to correctly add the dbt Logo into the aside while removing from the main navbar only on DCS pages on desktop. I've tried to find how to do this in the docusaurus documentation but no luck. An example we need to replicate is on the site https://ionicframework.com/docs/ where HTML code is embedded into the aside somehow. This is what we need to do on DOCS. **--FIXED**
2. **TOC 'edit this page' link:** We still need to add the 'edit this page' link into the bottom of the TOC. Hopefully Jason can help with this? **--FIXED**
3. **HOMEPAGE on Dark Mode:** So far this pull request can't go live until the new desktop version is complete. I can temporarily override the text styles in dark mode for now if you want the DOCS pages that I updated here to go live before the homepage style updates are complete. Emily, please let me know what you think on this? **--FIXED**

Preview Link
https://deploy-preview-1990--docs-getdbt-com.netlify.app/

### CODE REVIEW

**Comment by JKarlavige** [comment]
@zesuma Haven't done a full review yet, but have an idea for the blockers noted:

- For 1, It looks like Docusaurus adds the logo to the sidebar when the navbar `hideOnScroll` feature is enabled. [Here is the docs](https://docusaurus.io/docs/api/themes/configuration#auto-hide-sticky-navbar) on where to add it in the `docusaurus.config.js`. Not sure if hiding the nav on scroll was discussed, but seems to work well when testing locally. I'd recommend moving forward with this, then we won't need to set the logo with a pseudo class. Here is the component under the hood for you to see where the `hideOnScroll` is activating the logo [component in git](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/DocSidebar/Desktop/index.tsx)
- For 2 - We actually already have the `TOC` component swizzled under `website/src/theme/TOC/index.js`. This is where you can add the `Edit this page` link. I'm not sure what the requirements are for the Edit Page link or where it should point to, but you should be able to do everything needed within this `TOC` component.
- For 3 - It looks like the homepage update isn't due until Oct. 7th. Rather than holding up this task, my suggestion is to get the dark mode working for the homepage as-is within this PR. Then @breezyfasano can merge these edits once they are live to make any adjustments for the new homepage. I'll leave a note in Asana as well for this.

**Comment by zesuma** [comment]
> @zesuma Haven't done a full review yet, but have an idea for the blockers noted:
> 
> * For 1, It looks like Docusaurus adds the logo to the sidebar when the navbar `hideOnScroll` feature is enabled. [Here is the docs](https://docusaurus.io/docs/api/themes/configuration#auto-hide-sticky-navbar) on where to add it in the `docusaurus.config.js`. Not sure if hiding the nav on scroll was discussed, but seems to work well when testing locally. I'd recommend moving forward with this, then we won't need to set the logo with a pseudo class. Here is the component under the hood for you to see where the `hideOnScroll` is activating the logo [component in git](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/DocSidebar/Desktop/index.tsx)
> * For 2 - We actually already have the `TOC` component swizzled under `website/src/theme/TOC/index.js`. This is where you can add the `Edit this page` link. I'm not sure what the requirements are for the Edit Page link or where it should point to, but you should be able to do everything needed within this `TOC` component.
> * For 3 - It looks like the homepage update isn't due until Oct. 7th. Rather than holding up this task, my suggestion is to get the dark mode working for the homepage as-is within this PR. Then @breezyfasano can merge these edits once they are live to make any adjustments for the new homepage. I'll leave a note in Asana as well for this.

Thanks @JKarlavige ! This was very helpful! I have fixed each of these blockers. Thank you for your advice on these!

**Comment by zesuma** [comment]
Hi @JKarlavige, this is ready if you want for your QA: https://deploy-preview-1990--docs-getdbt-com.netlify.app/

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css]: This is a really creative solution to have the rest of the nav separate from the logo on docs pages. One thing I noticed, if i'm on a docs page where the sidebar is not used, the logo is missing altogether, [this page for example](https://deploy-preview-1990--docs-getdbt-com.netlify.app/community/maintaining-a-channel).

On the [docusaurus site](https://docusaurus.io/docs), the logo is still there in the same position, and when the navbar moves up on scroll the logo in the sidebar becomes visible.

Rather than hiding the nav logo altogether for docs pages, I wonder if we could set the width of logo link width to the width of the sidebar, which should push the rest of the nav over while keeping the logo in place. That way the logo would still exist in the main nav regardless if a sidebar exists or not.

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css]: The design has `20px`, so it's correct per the design, but I wonder if we should decrease the size of the text in the footer, it seems way big. `16px` or `18px` looks solid in my opinion, will leave it up to you!

File: website/docusaurus.config.js
Comment (JKarlavige) [inline website/docusaurus.config.js]: Are we removing these nav items for this task? I know there's a separate task to update the primary nav, but not sure if it was discussed to remove them here or not.

**Comment by JKarlavige** [comment]
On pages where the FAQ component is used, the styling does not update when in dark mode. Example: https://deploy-preview-1990--docs-getdbt-com.netlify.app/reference/dbt-jinja-functions/dispatch#faqs

This is a custom component, under `website/src/components/faqs`, with the stylesheet being in that directory.

image

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css:317]: The new sidebar is working great! The only thing we might want to do is indent subnav sidebar items. It could get confusing quick when opening a few sidebar items with everything aligned all the way to the left. I think following how the live site indents the subnav items would be sufficient for this. 

image
image

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css]: Same as the footer text, I feel like the paragraph text within the docs is slightly too big. It was originally 16px on the live site, and looks like it's at 20px in this preview. Maybe meeting in the middle at `18px` would be the sweet spot?

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css:744]: The blog breadcrumb styles do not update when switching to dark mode.

image

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css:921]: The CTA text on the cards turn dark when in dark mode. It feels like the text should remain white as the rest of the card styles do not change. This also occurs on the blog header card which uses the same styles.

image
image

File: website/src/css/custom.css
Comment (JKarlavige) [inline website/src/css/custom.css:452]: The "note" admonition styles do not update in dark mode. Example page: https://deploy-preview-1990--docs-getdbt-com.netlify.app/blog/change-data-capture

image

**Comment by JKarlavige** [comment]
The details/summary html tags do not update in dark mode, example page: https://deploy-preview-1990--docs-getdbt-com.netlify.app/guides/legacy/debugging-errors#invalid-ref-function (the "Debugging" dropdown.

**Comment by zesuma** [comment]
Hows this look?:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/reference/dbt-jinja-functions/dispatch#faqs

**Review by JKarlavige — CHANGES_REQUESTED** [review]
So far it's looking good! Have some requested changes and left some notes, mostly around colors not updating when in dark mode. Let me know if you have any questions!

File: website/docusaurus.config.js
Comment (runleonarun) [reply website/docusaurus.config.js]: Yea @JKarlavige We should probably hold off moving [dbt-cloud and dbt-cli ](https://github.com/dbt-labs/docs.getdbt.com/pull/1990/files#diff-bc4913d026ae51bef2bc8b7a1c040fdf52b999d99e6fa40d87b379f98a60e0eaL112-L123) until the content is moved. But "Guides" can definitely go under "Learn"

File: website/docusaurus.config.js
Comment (runleonarun) [inline website/docusaurus.config.js]: "Courses" below should become "Learn" per https://www.figma.com/file/FxfxSLnmL38WKHr1AR1LNT/DevHub-Top-Level-Navigation?node-id=12%3A393#258651098

Comment (runleonarun) [inline website/docusaurus.config.js]: "Developer blog" should also move under "Learn" (which currently says "Courses") per https://www.figma.com/file/FxfxSLnmL38WKHr1AR1LNT/DevHub-Top-Level-Navigation?node-id=12%3A393#258651098

**Review by runleonarun — DISMISSED** [review]
This is looking really great! I had some comments and a general dark mode question/nit. This could totally be me, but I feel like when in light mode, I should click a moon icon to turn it to dark mode. It feels weird to be clicking a sun in light mode that turns it dark. But maybe this is standard practice?

File: website/docusaurus.config.js
Comment (nghi-ly) [reply website/docusaurus.config.js]: if we wanna move "Guides" under "Learn" (fka "Courses") in this PR, does it make sense to move "Contributing" from the left sidebar under "Community" as well to get that part done.

or is it better to leave all nav changes in another PR? scope-wise, this might make sense given the video and figma in the Asana ticket.

**Review by nghi-ly — COMMENTED** [review]
the new design looks so amazing! left one comment and also +1 to what runleonarun mentioned re: the sun/moon icon

**Comment by JKarlavige** [comment]
> or is it better to leave all nav changes in another PR? scope-wise, this might make sense given the video and figma in the Asana ticket.

@nghi-ly @runleonarun yes that is correct, the navigation should not be updated in this task, as it's being updated within a separate task. I believe the navigation should be the same as the live site for completion of this task.

**Review by john-rock — COMMENTED** [review]
Hey this is looking great so far, nice work!

Just a few comments:
Can we make the `hash-link` lighter when it dark mode? 
What_is_dbt____dbt_Docs

Same thing for the FAQ chevrons
Tests___dbt_Docs

The code blocks title bar background color does not update in dark mode.
Documentation___dbt_Docs

There is no sidebar or the navbar does not extend all the way on https://deploy-preview-1990--docs-getdbt-com.netlify.app/community/maintaining-a-channel
Maintaining_a_Slack_channel___dbt_Docs

Should we add a little bit of top margin to the footer socials on mobile?
dbt_Docs


The active breadcrumb within docs is slightly difficult to read and is also failing sufficient contrast accessibility checks. Can/should we fix the contrast on the text?
Tests___dbt_Docs

**Comment by zesuma** [comment]
> https://deploy-preview-1990--docs-getdbt-com.netlify.app/reference/dbt-jinja-functions/dispatch#faqs

let me know if you need anything changed on this?:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/reference/dbt-jinja-functions/dispatch#faqs

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css]: @JKarlavige this should be working now:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/docs/introduction

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css]: @JKarlavige I changed the font size to 18px. Let me know if this works for you?
https://deploy-preview-1990--docs-getdbt-com.netlify.app/docs/introduction

File: website/docusaurus.config.js
Comment (zesuma) [reply website/docusaurus.config.js]: Hi all. I've added these items back so this should be fixed.

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css:317]: I've added the indents:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/docs/introduction

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css]: changed to 18px:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/docs/introduction

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css:744]: breadcrumbs should now look good on dark mode:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/docs/introduction

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css:921]: should be working now:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/

File: website/src/css/custom.css
Comment (zesuma) [reply website/src/css/custom.css:452]: should be working now:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/blog/change-data-capture

File: website/docusaurus.config.js
Comment (zesuma) [reply website/docusaurus.config.js]: Hi @runleonarun The items are back in the navbar:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/blog/change-data-capture

**Comment by zesuma** [comment]
> Hey this is looking great so far, nice work!
> 
> Just a few comments: Can we make the `hash-link` lighter when it dark mode? What_is_dbt____dbt_Docs
> 
> Same thing for the FAQ chevrons Tests___dbt_Docs
> 
> The code blocks title bar background color does not update in dark mode. Documentation___dbt_Docs
> 
> There is no sidebar or the navbar does not extend all the way on https://deploy-preview-1990--docs-getdbt-com.netlify.app/community/maintaining-a-channel Maintaining_a_Slack_channel___dbt_Docs
> 
> Should we add a little bit of top margin to the footer socials on mobile? dbt_Docs
> 
> The active breadcrumb within docs is slightly difficult to read and is also failing sufficient contrast accessibility checks. Can/should we fix the contrast on the text? Tests___dbt_Docs

Great finds @john-rock these changes should be updated now:
https://deploy-preview-1990--docs-getdbt-com.netlify.app/

**Comment by zesuma** [comment]
> 

Hi @runleonarun I had this discussion with @JKarlavige about this and we decided due to the complexity of how docusaurus sets up their default light/dark mode icons,  overriding the default styles is not ideal at this time so we decided to keep them as is for now.

**Comment by JKarlavige** [comment]
Hey @zesuma, working through the updates now. One small nitpick: The arrow for the version dropdown collapses under the version number as i shrink the navigation down. This has a custom `nav-versioning` class, so it should be easy to target with css. Could this be adjusted to keep the arrow on the same line as the version number?
image

File: website/src/css/custom.css
Comment (JKarlavige) [reply website/src/css/custom.css]: @zesuma The footer links still look a bit large, could they be shrunken down a bit?
image

**Review by JKarlavige — CHANGES_REQUESTED** [review]
The updates look great @zesuma! left comments for two additional requests, aside from that this is looking just about ready to go! 

Also, no rush to get those updates in place today, we'll hold off on merging until next week

**Comment by zesuma** [comment]
> Hey @zesuma, working through the updates now. One small nitpick: The arrow for the version dropdown collapses under the version number as i shrink the navigation down. This has a custom `nav-versioning` class, so it should be easy to target with css. Could this be adjusted to keep the arrow on the same line as the version number? image

**Comment by zesuma** [comment]
Hi @JKarlavige I came up with a solution for the version dropdown so this should be all set.

File: website/src/components/faqs/styles.module.css
Comment (JKarlavige) [inline website/src/components/faqs/styles.module.css:25]: Nice!! I like this

File: website/src/theme/TOC/index.js
Comment (JKarlavige) [inline website/src/theme/TOC/index.js:34]: Nice work getting this implemented!

**Review by JKarlavige — APPROVED** [review]
lgtm, great work!! :shipit:

**Pull request merged**
